libs/clients/work-machines/src/lib/workMachines.service.ts

```diff
@@ -4,33 +4,125 @@ import {
   ApiMachinesGetRequest,
   ExcelRequest,
   GetMachineRequest,
+  MachineCategoryApi,
   MachineHateoasDto,
+  MachineOwnerChangeApi,
   MachinesApi,
   MachinesDocumentApi,
   MachinesFriendlyHateaosDto,
 } from '../../gen/fetch'
+import {
+  MachineDto,
+  ChangeMachineOwner,
+  ConfirmOwnerChange,
+} from './workMachines.types'
+import {
+  apiChangeMachineOwnerToApiRequest,
+  confirmChangeToApiRequest,
+} from './workMachines.utils'
+import { CustomMachineApi } from './providers'
 
 @Injectable()
 export class WorkMachinesClientService {
   constructor(
     private readonly machinesApi: MachinesApi,
+    private readonly machineApi: CustomMachineApi,
     private readonly docApi: MachinesDocumentApi,
+    private readonly machineOwnerChangeApi: MachineOwnerChangeApi,
+    private readonly machineCategoryApi: MachineCategoryApi,
   ) {}
 
-  private apiWithAuth = (user: User) =>
+  private machinesApiWithAuth = (user: User) =>
     this.machinesApi.withMiddleware(new AuthMiddleware(user as Auth))
 
+  private machineApiWithAuth = (user: User) =>
+    this.machineApi.withMiddleware(new AuthMiddleware(user as Auth))
+
+  private machineOwnerChangeApiWithAuth(auth: Auth) {
+    return this.machineOwnerChangeApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  private machineCategoryApiWithAuth(auth: Auth) {
+    return this.machineCategoryApi.withMiddleware(new AuthMiddleware(auth))
+  }
   getWorkMachines = (
     user: User,
     input: ApiMachinesGetRequest,
   ): Promise<MachinesFriendlyHateaosDto> =>
-    this.apiWithAuth(user).apiMachinesGet(input)
+    this.machinesApiWithAuth(user).apiMachinesGet(input)
 
   getWorkMachineById = (
     user: User,
     input: GetMachineRequest,
-  ): Promise<MachineHateoasDto> => this.apiWithAuth(user).getMachine(input)
+  ): Promise<MachineHateoasDto> =>
+    this.machinesApiWithAuth(user).getMachine(input)
 
   getDocuments = (user: User, input: ExcelRequest): Promise<Blob> =>
     this.docApi.withMiddleware(new AuthMiddleware(user as Auth)).excel(input)
+
+  async getMachines(auth: User): Promise<MachineDto[]> {
+    const result = await this.machinesApiWithAuth(auth).apiMachinesGet({
+      onlyShowOwnedMachines: true,
+    })
+    return (
+      result?.value?.map((machine) => {
+        return {
+          id: machine.id,
+          type: machine.type || '',
+          category: machine?.category || '',
+          regNumber: machine?.registrationNumber || '',
+          status: machine?.status || '',
+        }
+      }) || []
+    )
+  }
+  public async getMachineDetail(auth: User, id: string): Promise<MachineDto> {
```

Comment:
~ For readability please add line between functions so the code is not as cluttered.
```suggestion
  }
  
  public async getMachineDetail(auth: User, id: string): Promise<MachineDto> {
```